kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/data/vector/DefaultViewVector.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.defaults.internal.data.vector;
+
+import java.math.BigDecimal;
+
+import io.delta.kernel.data.ArrayValue;
+import io.delta.kernel.data.ColumnVector;
+import io.delta.kernel.data.MapValue;
+import io.delta.kernel.data.Row;
+import io.delta.kernel.types.DataType;
+import static io.delta.kernel.defaults.internal.DefaultKernelUtils.checkArgument;
+
+/**
+ * Provides a restricted view on an underlying column vector.
+ */
+public class DefaultViewVector implements ColumnVector {
+
+    private final ColumnVector underlyingVector;
+    private final int offset;
+    private final int size;
+
+    /**
+     * @param underlyingVector the underlying column vector to read
+     * @param start the row index of the underlyingVector where we want this vector to start
+     * @param end the row index of the underlyingVector where we want this vector to end
+     *            (non-inclusive)
+     */
+    public DefaultViewVector(ColumnVector underlyingVector, int start, int end) {
+        this.underlyingVector = underlyingVector;
+        this.offset = start;
+        this.size = end - start;
+    }
+
+    @Override
+    public DataType getDataType() {
+        return underlyingVector.getDataType();
+    }
+
+    @Override
+    public int getSize() {
+        return size;
+    }
+
+    @Override
+    public void close() {
+        underlyingVector.close();
```

Comment:
This could be a problem. We shouldn't be closing the underlying vector as it could still be used.

For Java heap-based vectors this doesn't matter, for off-heap-based ones, we need some kind of ref counting.